app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt

```diff
@@ -386,5 +406,10 @@ fun Data.getMigrationFailure(): MigrationData.Result.Failure = when (this.getStr
         this.getString(MigrationData.Result.Failure.KEY_FAILURE_ERROR_CODE) ?: ""
     )
 
-    else -> MigrationData.Result.Failure.NoNetwork
+    MigrationData.Result.Failure.KEY_MIGRATION_EXCEPTION -> this.keyValueMap[MigrationData.Result.Failure.KEY_MIGRATION_EXCEPTION]?.let {
+        MigrationData.Result.Failure.Unknown(it as? Throwable)
+    } ?: MigrationData.Result.Failure.Unknown(null)
+
+    null -> MigrationData.Result.Failure.Unknown(null)
+    else -> MigrationData.Result.Failure.Unknown(null)
```

Comment:
In the line 391 an empty `workDataOf()` is passed for any other type, and it that case it was only `NoNetwork` type left so empty `workData` meant it was a `NoNetwork`. Maybe provide another `KEY_FAILURE_TYPE` for `NoNetwork` types, because now, on the other hand, `NoNetwork` is not being returned at all 😅